fix!: Align OAuth Application specs with API

chanioxaris opened the pull request:
In this commit we are making changes to the OAuth Application resource API. We are aligning the types of the Create operation, to optional, as the only required right now is the 'name' property Also we are exposing a new 'token_introspection_url' property as part of the response. Those are considered breaking changes, that's why they only affect the new major version (v3)

kostaspt commented on oauth_application.go:
```diff
@@ -2,21 +2,22 @@ package clerk
+	TokenIntrospectionURL string  `json:"token_introspection_url"`
+	CreatedAt             int64   `json:"created_at"`
+	UpdatedAt             int64   `json:"updated_at"`
+	ClientSecret          *string `json:"client_secret,omitempty"`
```
🙃 (nit) Maybe move this closer to `ClientID`?

chanioxaris replied:
Good call out!

kostaspt commented on oauthapplication/client_test.go:
```diff
@@ -36,9 +36,9 @@ func TestOAuthApplicationClientCreate(t *testing.T) {
-		Scopes:      scopes,
-		Public:      public,
+		CallbackURL: clerk.String(callbackURL),
+		Scopes:      clerk.String(scopes),
```
🙃 What if we forced the scopes to be a `[]string`, since it's a breaking change, and then joined them when creating the params for the Backend API?

chanioxaris replied:
Yeah definitely agree here! It's on my list for the next major API version

chanioxaris committed:
fix!: Align OAuth Application specs with API

In this commit we are making changes to the OAuth Application resource API. We are aligning the
types of the Create operation, to optional, as the only required right now is the 'name' property
Also we are exposing a new 'token_introspection_url' property as part of the response.
Those are considered breaking changes, that's why they only affect the new major version (v3)
